torch/csrc/StorageSharing.cpp

```diff
@@ -616,7 +616,8 @@ PyObject* THPStorage_sharedFd(PyObject* self, PyObject* noargs) {
 
 PyObject* THPStorage_isShared(PyObject* self, PyObject* noargs) {
   const auto& storage = THPStorage_Unpack(self);
-  if (storage.device_type() == at::kCUDA) {
+  if (storage.device_type() == at::kCUDA ||
+      storage.device_type() == at::kPrivateUse1) {
```

Comment:
Why should storage.is_shared() return True for kPrivateUse1?

Reply:
I think the best thing here would be to ask the user-defined allocator if the current storage is shared or not.
That will give you full flexibility for the different backends you might have.

Reply:
yeah, I will have a check and test what you said. And I add warning info if we use `torch.device(0)` or `torch.rand(2, device=0)`, we suggest to use `torch.device("cuda:0")` or `torch.rand(2,device="cuda:0")`. Could you have a look? @albanD